src/rviz/TrajectoryMarker.cpp

```diff
@@ -0,0 +1,259 @@
+void TrajectoryMarker::setColor(const Eigen::Vector3d& rgb)
+{
+  auto& marker = getMarker();
+  marker.color.r = rgb[0];
+  marker.color.g = rgb[1];
+  marker.color.b = rgb[2];
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+Eigen::Vector3d TrajectoryMarker::getColor() const
+{
+  const auto& marker = getMarker();
+  return convertROSColorRGBAToEigen(marker.color).head<3>();
+}
+
+//==============================================================================
+void TrajectoryMarker::setAlpha(double alpha)
+{
+  auto& marker = getMarker();
+  marker.color.a = alpha;
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+double TrajectoryMarker::getAlpha() const
+{
+  const auto& marker = getMarker();
+  return marker.color.a;
+}
+
+//==============================================================================
+void TrajectoryMarker::setRGBA(const Eigen::Vector4d& rgba)
+{
+  auto& marker = getMarker();
+  marker.color = convertEigenToROSColorRGBA(rgba);
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+Eigen::Vector4d TrajectoryMarker::getRBGA() const
+{
+  const auto& marker = getMarker();
+  return convertROSColorRGBAToEigen(marker.color);
+}
+
+//==============================================================================
+void TrajectoryMarker::setThickness(double thickness)
+{
+  auto& marker = getMarker();
+  marker.scale.x = thickness;
+
+  mNeedUpdate = true;
+}
+
+//==============================================================================
+double TrajectoryMarker::getThickness() const
+{
+  const auto& marker = getMarker();
+  return marker.scale.x;
+}
+
+//==============================================================================
+void TrajectoryMarker::setNumLineSegments(std::size_t numLineSegments)
+{
+  if (numLineSegments < 2)
+  {
+    throw std::invalid_argument(
+        "Number of line segments should be greater than 2");
+  }
+
+  mNumLineSegments = numLineSegments;
+
+  mNeedPointsUpdate = true;
+}
+
+//==============================================================================
+std::size_t TrajectoryMarker::getNumLineSegments() const
+{
+  return mNumLineSegments;
+}
+
+//==============================================================================
+void TrajectoryMarker::update()
+{
+  if (mNeedPointsUpdate)
+    updatePoints();
+
+  if (!mNeedUpdate)
+    return;
+
+  mMarkerServer->insert(mInteractiveMarker);
+
+  mNeedUpdate = false;
+}
+
+//==============================================================================
+void TrajectoryMarker::updatePoints()
+{
+  using visualization_msgs::Marker;
+
+  if (!mNeedPointsUpdate)
+    return;
+
+  auto& marker = getMarker();
+  auto& points = marker.points;
+  points.clear();
+
+  if (!mTrajectory)
+    return;
+
+  Eigen::VectorXd savedPositions = mSkeleton.getPositions();
```

Comment:
Is there a reason for using this over `MetaSkeletonSaver`?

Reply:
That's because `MetaSkeletonSaver` saves more than positions, but I'm open to using `MetaSkeletonSaver`.

Reply:
I think the old OpenRAVE state saver allowed you to pass in flags to that specified what should be saved. Can you add an issue for supporting that in `MetaSkeletonStateSaver`, and use `MetaSkeletonStateSaver` here?

Reply:
Done (https://github.com/personalrobotics/aikido/issues/293).